controller/src/api/external/routes/tenant.rs

```diff
@@ -1,29 +1,32 @@
 use route_recognizer;
 use rusqlite::Connection;
-use std::io;
 use std::str::FromStr;
 use std::sync::mpsc::Sender;
 use tracing::{event, Level};
 
-use crate::api;
+use super::HttpResult;
 use crate::api::external::services::element::elements_set_right_name;
 use crate::api::types::element::OnlyId;
 use crate::api::types::tenant::Tenant;
-use crate::api::ApiChannel;
+use crate::api::{ApiChannel, RikError};
 use crate::database::RikRepository;
 
 pub fn get(
     _: &mut tiny_http::Request,
     _: &route_recognizer::Params,
     connection: &Connection,
     _: &Sender<ApiChannel>,
-) -> Result<tiny_http::Response<io::Cursor<Vec<u8>>>, api::RikError> {
+) -> HttpResult {
     if let Ok(mut tenants) = RikRepository::find_all(connection, "/tenant") {
         tenants = elements_set_right_name(tenants.clone());
-        let tenants_json = serde_json::to_string(&tenants).unwrap();
+        let tenants_json = serde_json::to_string(&tenants).map_err(RikError::ParsingError)?;
```

Comment:
In order to have a deserialization error that is more understandable for the user, we could try to use [serde_path_to_error](https://docs.rs/serde_path_to_error/latest/serde_path_to_error/) crate which will give a complete path to what is the problem

Reply:
Good idea to add [serde_path_to_error](https://docs.rs/serde_path_to_error/latest/serde_path_to_error/) but in another PR #120